src/clang-cpp-frontend/clang_cpp_adjust_code.cpp

```diff
@@ -127,3 +139,51 @@ void clang_cpp_adjust::adjust_for(codet &code)
   else
     clang_c_adjust::adjust_for(code);
 }
+
+void clang_cpp_adjust::adjust_decl_block(codet &code)
+{
+  codet new_block("decl-block");
+
+  Forall_operands(it, code)
+  {
+    if(it->is_code() && (it->statement() == "skip"))
+      continue;
+
+    code_declt &code_decl = to_code_decl(to_code(*it));
+
+    if(code_decl.operands().size() == 2)
+    {
+      exprt &rhs = code_decl.rhs();
+      if(
+        rhs.id() == "sideeffect" && rhs.statement() == "function_call" &&
+        rhs.get_bool("constructor"))
+      {
+        // turn struct BLAH bleh = BLAH() into two instructions:
+        // struct BLAH bleh;
+        // BLAH(&bleh);
+
+        // First, create new decl without rhs
+        code_declt object(code_decl.lhs());
+        new_block.copy_to_operands(object);
+
+        // Get rhs
+        side_effect_expr_function_callt &init =
+          to_side_effect_expr_function_call(rhs);
+
+        init.arguments().push_back(address_of_exprt(code_decl.lhs()));
```

Comment:
Add a comment here to keep the standard of the other lines?

Reply:
Done